src/map.cpp

```diff
@@ -5123,129 +5113,27 @@ std::list<item> map::use_charges( const tripoint &origin, const int range,
             continue;
         }
 
-        const cata::optional<vpart_reference> kpart = vp.part_with_feature( "FAUCET", true );
```

Comment:
Okay, you're here and I didn't ask you to go play in this space, but you did and so you get the cleanup.

This is an awful function, it's still an awful function after the cleanup.  90% of this nonsense should be moved into `src/vehicle_use.cpp` as a separate function that map just calls the way the basecamp stuff is a call into the basecamp code.  This CS 101 abstraction stuff.  I've been meaning to do it for a while but keep getting busy, but you're touching the code so it's on you to clean it up.

Reply:
Expanding on this comment as I look at the code again:
```diff
diff --git a/src/map.cpp b/src/map.cpp
index 7daf008a0a0..443de300b02 100644
--- a/src/map.cpp
+++ b/src/map.cpp
@@ -5119,8 +5119,11 @@ std::list<item> map::use_charges( const tripoint &origin, const int range,
         }

         const optional_vpart_position vp = veh_at( p );
-        if( !vp ) {
-            continue;
+        if( vp ) {
+            ret = vp->use_charges( type, quantity );
+           if( quantity <= 0 ) {
+                return ret;
+            }
         }

         const cata::optional<vpart_reference> kpart = vp.part_with_feature( "FAUCET", true );

```
then everything starting at old line 5126 gets move into `src/vehicle_use.cpp` as part of `vehicle::use_charges()` and you do whatever else you have to do inside the vehicle code where it will be 100% cleaner.  It looks like you have the logic right, just need to relocate the code.  It is going to be increase the size of your diff but not by all that much since you're changing the relevant code anyway.

Reply:
Also when you have time - could you peek at https://github.com/CleverRaven/Cataclysm-DDA/pull/44735/files and tell if this is what you had in mind... I swear there was a button somewhere to request a review rather than pester manually but...